packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue

```diff
@@ -375,12 +339,66 @@ export default defineComponent({
       return l1Direct ? -1 : 1
     },
 
+    checkPasswordEnforcedFor(link) {
+      const currentRole = this.availableRoleOptions.find(({ role }) => {
```

Comment:
There is `LinkShareRoles.getByBitmask` which feels more appropriate than bitmask comparisons. But I guess filtering the `availableRoleOptions` was on purpose? Do you see a way to incorporate the `getByBitmask` method anyway?

Reply:
No clue where&why the mapping in the line below ever was introduced in the past (we found, disliked but accepted&reused the code previously found). I went for your suggested solution and seems to work like a charm